docs/rules/jsx-no-bind.md

```diff
@@ -2,7 +2,7 @@
 
 <!-- end auto-generated rule header -->
 
-A `bind` call or [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) in a JSX prop will create a brand new function on every single render. This is bad for performance, as it may cause unnecessary re-renders if a brand new function is passed as a prop to a component that uses reference equality check on the prop to determine if it should update.
+Using `bind` on a function or declaring a function in the render method of a component, and then passing that function as a prop will mean a brand new function is created on every single render. This can affect performance in some situations, as it may cause unnecessary re-renders if a brand new function is passed as a prop to a component that uses reference equality check on the prop to determine if it should update, such as a component wrapped with [`memo`](https://react.dev/reference/react/memo#memo).
```

Comment:
```suggestion
Using `bind` on a function or declaring a function in the render method of a component or the body of a functional component, and then passing that function as a prop will mean that the brand new function that is created on every single render will be considered a completely different function. This can affect performance in some situations, as it may cause unnecessary re-renders if a brand new function is passed as a prop to a component that uses reference equality check on the prop to determine if it should update, such as a component wrapped with [`memo`](https://react.dev/reference/react/memo#memo), or if the prop is used in any hook's "dependency array".
```

Reply:
I think the current correct and incorrect usage examples clearly demonstrate all the cases I'm aware of, but I thought all possibilities should be referenced in the rationale as well, which I think they now are.

I think that calling out the dependency array is a good example of the performance impact but from looking into it it doesn't seem like it's covered by the rule. I think that could create ambiguity about what the rule covers, even it's useful information, what do you think? Am I missing anything with that?

Reply:
I don't think it can - hook dependency arrays is just one of the ways that creating a function inside the render path, and passing it around, can cause issues.

Reply:
Makes sense, well I'm happy to go ahead with that 👍